File: python_backend/fenParser/getFen/get_digits.py


def get_digits(fen):
    """getFen TASK 5, replace consecutive '#' (empty squares) with the digit for how many"""
    for consect in ['########', '#######', '######', '#####', '####', '###', '##', '#']:
        fen = fen.replace(consect, str(len(consect)))
    return fen


if __name__ == "__main__":

    # test 1
    print('test 1')
    print(get_digits('rnbqkbnr/pppppppp/########/########/########/########/PPPPPPPP/RNBQKBNR'))
    print('\n')

    # test 2:
    print('test 2')
    print(get_digits('r###k##r/#P######/#####P##/p###n##p/###b#BpP/NBQb#q##/########/R###K##R'))
    print('\n')

    # test 3:
    print('test 3')
    print(get_digits('###k###b/q##r##P#/###P#P##/########/q#QK#QBr/##NP####/###R#R##/b##q##q#'))
    print('\n')

File: app (1)/src/game_logic/fenParser/getFen/getDigits.js


export function getDigits(fen) {
    /*getFen TASK 5, replace consecutive '#' (empty squares) with the digit for how many**/
    let consects = [/########/g, /#######/g, /######/g, /#####/g, /####/g, /###/g, /##/g, /#/g]
    let consectLens = [8, 7, 6, 5, 4, 3, 2, 1]
    for (let i = 0; i < consects.length; i++) {
        fen = fen.replace(consects[i], consectLens[i].toString());
    }
    return fen;
}

// module.exports = getDigits;

/** for node.js
if (require.main === module) {
    
    // test 1
    console.log('test 1')
    console.log(getDigits('rnbqkbnr/pppppppp/########/########/########/########/PPPPPPPP/RNBQKBNR'))
    console.log('\n')

    // test 2:
    console.log('test 2')
    console.log(getDigits('r###k##r/#P######/#####P##/p###n##p/###b#BpP/NBQb#q##/########/R###K##R'))
    console.log('\n')

    // test 3:
    console.log('test 3')
    console.log(getDigits('###k###b/q##r##P#/###P#P##/########/q#QK#QBr/##NP####/###R#R##/b##q##q#'))
    console.log('\n')

}
*/
